lib/Controller/PageController.php

```diff
@@ -86,7 +86,10 @@ public function index(): TemplateResponse {
 		$this->initialStateService->provideInitialState($this->appName, 'image-mimes', Application::IMAGE_MIMES);
 		$this->initialStateService->provideInitialState($this->appName, 'video-mimes', Application::VIDEO_MIMES);
 		$this->initialStateService->provideInitialState($this->appName, 'maps', $this->appManager->isEnabledForUser('maps') === true);
+
+		// User Settings
 		$this->initialStateService->provideInitialState($this->appName, 'croppedLayout', $this->config->getUserValue($user->getUid(), Application::APP_ID, 'croppedLayout', 'false'));
+		$this->initialStateService->provideInitialState($this->appName, 'timelineRootFolder', $this->config->getUserValue($user->getUid(), Application::APP_ID, 'timelineRootFolder', '/'));
```

Comment:
We should probably limit the allowed keys here: https://github.com/nextcloud/photos/blob/441f97020bee10aeb9e0b27cd2ac174ed02dc3b1/lib/Controller/ApiController.php#L66

As a safety mesure :thinking:

Reply:
I'm not sure what you mean

Reply:
setUserConfig allows anything, and will put it straight into the db.
Meaning we could override existing values. We should check if the $key is either `timelineRootFolder` or `croppedLayout` and throw otherwise as a safety measure

Reply:
ok gotcha